src/Native/CMakeLists.txt

```diff
@@ -24,6 +24,16 @@ if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
     add_compile_options(-Wno-unreachable-code)
 endif ()
 
+if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+    add_definitions(-D_LINUX_)
+elseif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    add_definitions(-D_OSX_)
+elseif (CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
```

Comment:
There are platform default defines that we have been using everywhere else so far ... `__FreeBSD__`, `__NetBSD__`, ... . It is not a good idea to define your own for consistency...

I think it would be best to define the name in configure.cmake. Add something like this to configure.cmake:

```
if (CMAKE_SYSTEM_NAME STREQUAL Linux)  
    set(PAL_UNIX_NAME "LINUX")
elseif (CMAKE_SYSTEM_NAME STREQUAL Darwin)  
    set(PAL_UNIX_NAME "OSX")
...
```

Add this to `pal_config.h.in`:

`#cmakedefine PAL_UNIX_NAME @PAL_UNIX_NAME@`

And then just implement `SystemNative_GetUnixName()` as `return PAL_UNIX_NAME` without any ifdefs.